pkg/platform/runtime/runtime.go

```diff
@@ -177,27 +178,23 @@ func (r *Runtime) recordCompletion(err error) {
 		r.recordUsage()
 	}
 
-	var projectID string
-	if pj, err := model.FetchProjectByName(r.target.Owner(), r.target.Name()); err == nil {
-		projectID = pj.ProjectID.String()
-	} else {
-		multilog.Error("Unable to fetch project by name: %v", errs.JoinMessage(err))
-	}
-
 	ns := project.Namespaced{
 		Owner:   r.target.Owner(),
 		Project: r.target.Name(),
 	}
 
-	var errorType string
+	errorType := "unknown"
 	switch {
+	// IsInputError should always be first because it is technically possible for something like a
+	// download error to be cause by an input error.
 	case locale.IsInputError(err):
 		errorType = "input"
-	case errs.Matches(err, &setup.ProgressReportError{}): // should come before non-input errors
+	// Progress errors should come before any other non-input error.
```

Comment:
Please explain why (in the comment).

Reply:
Good that you had me do that. These should actually come last :/ Sorry about that.